<h1>Table of Contents<span class="tocSkip"></span></h1>
<div class="toc"><ul class="toc-item"><li><span><a href="#Описание-задания." data-toc-modified-id="Описание-задания.-1"><span class="toc-item-num">1&nbsp;&nbsp;</span>Описание задания.</a></span></li><li><span><a href="#Описание-данных." data-toc-modified-id="Описание-данных.-2"><span class="toc-item-num">2&nbsp;&nbsp;</span>Описание данных.</a></span></li><li><span><a href="#Предобработка-данных." data-toc-modified-id="Предобработка-данных.-3"><span class="toc-item-num">3&nbsp;&nbsp;</span>Предобработка данных.</a></span><ul class="toc-item"><li><span><a href="#Проверка-на-пропуски" data-toc-modified-id="Проверка-на-пропуски-3.1"><span class="toc-item-num">3.1&nbsp;&nbsp;</span>Проверка на пропуски</a></span></li><li><span><a href="#Проверка-на-аномалии-и-изучение-особенностей-в-датасете" data-toc-modified-id="Проверка-на-аномалии-и-изучение-особенностей-в-датасете-3.2"><span class="toc-item-num">3.2&nbsp;&nbsp;</span>Проверка на аномалии и изучение особенностей в датасете</a></span></li><li><span><a href="#Проверка-на-дубликаты" data-toc-modified-id="Проверка-на-дубликаты-3.3"><span class="toc-item-num">3.3&nbsp;&nbsp;</span>Проверка на дубликаты</a></span></li><li><span><a href="#Фильтрация-датасета" data-toc-modified-id="Фильтрация-датасета-3.4"><span class="toc-item-num">3.4&nbsp;&nbsp;</span>Фильтрация датасета</a></span></li><li><span><a href="#Проверка-процента-потерянных-данных" data-toc-modified-id="Проверка-процента-потерянных-данных-3.5"><span class="toc-item-num">3.5&nbsp;&nbsp;</span>Проверка процента потерянных данных</a></span></li><li><span><a href="#Изучение-перечня-скиллов" data-toc-modified-id="Изучение-перечня-скиллов-3.6"><span class="toc-item-num">3.6&nbsp;&nbsp;</span>Изучение перечня скиллов</a></span></li><li><span><a href="#Изучение-формата-работы" data-toc-modified-id="Изучение-формата-работы-3.7"><span class="toc-item-num">3.7&nbsp;&nbsp;</span>Изучение формата работы</a></span></li><li><span><a href="#Формирование-датасета-для-выгрузки" data-toc-modified-id="Формирование-датасета-для-выгрузки-3.8"><span class="toc-item-num">3.8&nbsp;&nbsp;</span>Формирование датасета для выгрузки</a></span></li></ul></li><li><span><a href="#Выводы-по-предобработке-данных" data-toc-modified-id="Выводы-по-предобработке-данных-4"><span class="toc-item-num">4&nbsp;&nbsp;</span>Выводы по предобработке данных</a></span></li><li><span><a href="#Дашборд-TableaU" data-toc-modified-id="Дашборд-TableaU-5"><span class="toc-item-num">5&nbsp;&nbsp;</span>Дашборд TableaU</a></span></li></ul></div>

# Hakaton Linked In

## Описание задания.

LinkedIn рестарт  
Проект: Анализ вакансии для аналитиков данных  
Цель: Визуализировать информацию о рынке вакансий для аналитиков в Европе  

Подготовка данных ( 2 вариант)  
Необходимо выполнить предобработку данных и подготовить датасет для работы с помощью BI инструментов.  
Определить требуемые хард скилы и тип занятости (online, hybride, on-site) для каждой вакансии.

## Описание данных.

Датасет содержит выгрузку с платформы Linked In содержащую информацию по различным вакансиям связанным с IT сектором.

## Предобработка данных.

Загрузим основные библиотеки для работы с данными.

In [1]:
import pandas as pd
import pprint as pp
from functools import reduce
import matplotlib.pyplot as plt
import plotly.express as px
from plotly import graph_objects as go
import seaborn as sns
import warnings
warnings.filterwarnings("ignore")

In [2]:
pd.set_option('display.max_columns', None) 
pd.set_option('display.max_rows', None)

Создадим переменную с датасетом.

In [3]:
dataset = pd.read_csv('C:/Users/truel/datasets/linkedin_ ver_2.0.csv', index_col=0)

Теперь посмотрим на датасет и проведем необходимую предобработку данных.

In [4]:
dataset.info()

<class 'pandas.core.frame.DataFrame'>
Index: 998 entries, Data Analyst to Analista de datos BI
Data columns (total 9 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   location         998 non-null    object 
 1   country          998 non-null    object 
 2   employment_type  998 non-null    object 
 3   company_name     996 non-null    object 
 4   employee_qty     998 non-null    object 
 5   company_field    998 non-null    object 
 6   skills           998 non-null    object 
 7   job_description  998 non-null    object 
 8   applicants       838 non-null    float64
dtypes: float64(1), object(8)
memory usage: 78.0+ KB


Практически во всех столбцах тип данных - object.

In [5]:
dataset.head()

location          country  \
title                                                            
Data Analyst                            Basel      Switzerland   
Data Analyst - Logistics             Coventry   United Kingdom   
Data Analyst - Logistics             Coventry   United Kingdom   
Data Analyst (Space & Planning)  South Molton   United Kingdom   
Data Analyst                           Lugano      Switzerland   

                                employment_type  \
title                                             
Data Analyst                            On-site   
Data Analyst - Logistics                On-site   
Data Analyst - Logistics                On-site   
Data Analyst (Space & Planning)         On-site   
Data Analyst                            On-site   

                                                             company_name  \
title                                                                       
Data Analyst                     PharmiWeb.Jobs: Global Life Science Jobs   
Data Analyst - Logistics                             Resolute Recruitment   
Data Analyst - Logistics                             Resolute Recruitment   
Data Analyst (Space & Planning)                       Mole Valley Farmers   
Data Analyst                                                      FORFIRM   

                                    employee_qty            company_field  \
title                                                                       
Data Analyst                     11-50 employees  Staffing and Recruiting   
Data Analyst - Logistics           not specified            not specified   
Data Analyst - Logistics           not specified            not specified   
Data Analyst (Space & Planning)    not specified            not specified   
Data Analyst                       not specified            not specified   

                                skills  \
title                                    
Data Analyst                      None   
Data Analyst - Logistics          None   
Data Analyst - Logistics          None   
Data Analyst (Space & Planning)   None   
Data Analyst                      None   

                                                                   job_description  \
title                                                                                
Data Analyst                                                 What You Will Achi...   
Data Analyst - Logistics                                                             
Data Analyst - Logistics                                      Data Analyst - Lo...   
Data Analyst (Space & Planning)                                Salary:     To b...   
Data Analyst                                                   FORFIRM     is p...   

                                 applicants  
title                                        
Data Analyst                           47.0  
Data Analyst - Logistics                NaN  
Data Analyst - Logistics                NaN  
Data Analyst (Space & Planning)         NaN  
Data Analyst                            NaN

### Проверка на пропуски

Проверим датасет на пропуски с помощью метода isna().sum()

In [6]:
dataset.isna().sum()

location             0
country              0
employment_type      0
company_name         2
employee_qty         0
company_field        0
skills               0
job_description      0
applicants         160
dtype: int64

Наибольшее количество пропусков присутствует в столбце applicants(заявители на вакансии). Вероятно данные пропуски можно объяснить тем что отбор по вакансии еще не завершен и количество заявителей не подсчитано. Также пропуски присутствуют в столбце company_name. Эти пропуски не критичны, поэтому можно не заполнять их.

### Проверка на аномалии и изучение особенностей в датасете

Воспользуемся методом describe и поищем интересные особенности и аномалии в данных.

In [7]:
dataset['location'].describe()

count       998
unique      428
top       Milan
freq         40
Name: location, dtype: object

Из 998 представленных вакансий - 428 уникальных локаций. Самая популярная локация - Милан.

In [8]:
dataset['country'].describe()

count        998
unique        72
top        Italy
freq         148
Name: country, dtype: object

В столбце country - 998 уникальных значения. Самая часто упоминаемая страна - Италия.

Воспользуемся методом value_counts и посмотрим на уникальные значения в столбце country.

In [9]:
dataset['country'].value_counts()

 Italy                                         148
 Germany                                       142
 France                                        114
 United Kingdom                                 89
 Netherlands                                    65
 Spain                                          54
 Poland                                         43
 Belgium                                        40
 Czechia                                        25
 Sweden                                         22
Rotterdam and The Hague                         21
 Portugal                                       21
 Luxembourg                                     14
 Greece                                         14
 Ireland                                        14
Germany                                         13
Poland                                          12
 Hungary                                        11
 Bulgaria                                       10
 Austria                       

Помимо названия стран в столбце присутствуют значения - названия географических локаций. Необходимо вручную заменить значения для дальнейшего удобства работы. Воспользуемся методом replace.

In [10]:
d = {'Brussels Metropolitan Area':'Belgium',
'Greater Paris Metropolitan Region':'France',
'Wroclaw Metropolitan Area':'Poland',
'Iasi Metropolitan Area':'Romania',
'Cologne Bonn Region':'Germany',
'Greater Munster Area':'Ierland',
'Greater Banska Bystrica Area':'Slovakia',
'Greater Nuremberg Metropolitan Area':'Germany',
'Berlin Metropolitan Area':'Germany',
'Greater Barcelona Metropolitan Area':'Spain',
'Rotterdam and The Hague':'Netherlands',
'Prague Metropolitan Area':'Czechia',
'Greater Pau Area':'France',
'Lisbon Metropolitan Area':'Portugal',
'Athens Metropolitan Area':'Greece',
'Stuttgart Region':'Germany',
'Greater Milan Metropolitan Area':'Italy',
'Amsterdam Area':'Netherlands',
'Eindhoven Area':'Netherlands',
'Krakow Metropolitan Area':'Poland',
'Greater Munich Metropolitan Area':'Germany',
'Greater Palma de Mallorca Metropolitan Area':'Spain',
'Warsaw Metropolitan Area':'Poland',
'Greater Verona Metropolitan Area':'Italy',
'Greater Madrid Metropolitan Area':'Spain',
'Greater Oslo Region':'Norway',
'Greater Dijon Area':'France',
'Greater Lyon Area':'France',
'Copenhagen Metropolitan Area':'Denmark'}


dataset['country'] = dataset['country'].replace(d)


In [11]:
dataset['country'].value_counts()

 Italy             148
 Germany           142
 France            114
 United Kingdom     89
 Netherlands        65
 Spain              54
 Poland             43
 Belgium            40
 Czechia            25
Netherlands         25
 Sweden             22
 Portugal           21
Germany             20
Poland              17
 Ireland            14
 Greece             14
 Luxembourg         14
Italy               11
 Hungary            11
 Bulgaria           10
 Austria             9
Spain                9
France               8
 Romania             7
Portugal             5
 Latvia              5
 Malta               5
 Denmark             4
 Slovakia            4
 Switzerland         4
Belgium              4
 Lithuania           4
Bulgaria             4
 Norway              3
 Finland             3
 Croatia             3
Romania              2
Ireland              2
Greece               2
Denmark              2
 Estonia             2
Ierland              1
 Monaco              1
Czechia    

Обрежем пробелы с помощью метода str.strip()

In [12]:
dataset['country'] = dataset['country'].str.strip()

In [13]:
dataset['country'].value_counts()

Germany           162
Italy             159
France            122
Netherlands        90
United Kingdom     89
Spain              63
Poland             60
Belgium            44
Portugal           26
Czechia            26
Sweden             22
Ireland            16
Greece             16
Bulgaria           14
Luxembourg         14
Hungary            12
Austria             9
Romania             9
Denmark             6
Lithuania           5
Latvia              5
Malta               5
Slovakia            5
Norway              4
Finland             4
Switzerland         4
Croatia             3
Estonia             2
Ierland             1
Monaco              1
Name: country, dtype: int64

In [14]:
dataset['employment_type'].describe()

count        998
unique         4
top       Hybrid
freq         486
Name: employment_type, dtype: object

В столбце employment_type описан режим работы для вакансии. Самый популярный формат работы - гибридный.

In [15]:
dataset['employment_type'].unique()

array(['On-site', 'Hybrid', 'not specified', 'Remote'], dtype=object)

On-site - офисный формат работы.  
Hybrid - гибридный формат работы.  
not specified - формат работы не указан.  
Remote - удаленный формат работы.  

In [16]:
dataset['company_name'].describe()

count           996
unique          669
top       CPM Italy
freq             45
Name: company_name, dtype: object

В датасете представлены 996 компаний из них уникальных 669. Самая часто упоминаемая компания - CPM Italy. В среднем у каждой компании 45 вакансий.

In [17]:
dataset['employee_qty'].describe()

count                   998
unique                   19
top       10,001+ employees
freq                    273
Name: employee_qty, dtype: object

В столбце employee_qty указано количество сотрудников в компании. Всего представлено 19 уникальных значений. Самое популярное свыше 10000 сотрудников.

In [18]:
dataset['company_field'].describe()

count                               998
unique                              122
top       IT Services and IT Consulting
freq                                153
Name: company_field, dtype: object

company_field - сфера деятельности компании. Всего 122 уникальных значения. Самое популярное - IT Services and IT Consulting. 

In [19]:
dataset['skills'].describe()

count      998
unique     617
top       None
freq       128
Name: skills, dtype: object

Столбец skills содержит описание необходимых навыков для вакансии. Всего 617 уникальных значений, самое популярное значение не определено - none.

In [20]:
dataset['job_description'].describe()

count                                                   998
unique                                                  812
top                                          This job is...
freq                                                     27
Name: job_description, dtype: object

Столбец job_description содержит описание вакансии, всего 812 уникальных значений. 

In [21]:
dataset['applicants'].describe()

count    838.000000
mean      58.052506
std       47.534251
min        1.000000
25%       22.000000
50%       45.000000
75%       86.000000
max      199.000000
Name: applicants, dtype: float64

В столбце applicants содержится количество заявителей на вакансию. Среднее значение 58 человек на 1 позицию. 

Проверим список уникальных профессий в вакансиях с помощью метода unique(). Так как название вакансии прописано в индексе то необходимо прописать это в команде.

In [22]:
dataset.index.unique()

Index(['Data Analyst', 'Data Analyst - Logistics',
       'Data Analyst (Space & Planning)', 'Data Analyst - Hybrid Working',
       'Commercial Data Analyst', 'Asset Data Analyst',
       'Data Analyst - Hybrid', 'Data Analyst (FT)', 'Data Analyst H/F',
       'Data Analyst II',
       ...
       'Marketing Analyst (m/w/d)', 'Data Analyst Job In Overseas',
       'CRO /Digital analyst', 'Customer Support Analyst', 'Marketing Analyst',
       'Emerton Data: Data Scientist & Strategy Consultant',
       'Ingeniero de datos Power BI + Azure Datafactory',
       'Business-Analyst:in', 'Junior Software Developer',
       'Analista de datos BI'],
      dtype='object', name='title', length=560)

Всего выявлено 560 уникальных названий вакансий.

### Проверка на дубликаты

Выполним проверку на дубликаты с помощью метода duplicated().sum()

In [23]:
dataset.duplicated().sum()

112

В датасете содержится 112 дубликатов. Необходимо удалить дубликаты с помощью метода drop_duplicates().reset_index(drop=False). В значение drop указываем False,для того чтобы перезаписать значения индекса в отдельных столбец title.

In [24]:
dataset = dataset.drop_duplicates().reset_index(drop=False)

### Фильтрация датасета

В соответствии с ТЗ нам необходимо оставить только вакансии в которых содержатся следующие наименования - Data Analyst|Data Scientist. Напишем функцию, которая отфильтрует датасет и оставить только необходимые нам вакансии.

In [25]:
def filter_jobs_by_title(dataset):
    filtered_dataset = dataset[dataset['title'].str.contains('Data Analyst|Data Scientist')]

    return filtered_dataset

filtered_dataset = filter_jobs_by_title(dataset)

In [26]:
filtered_dataset.info()

<class 'pandas.core.frame.DataFrame'>
Int64Index: 426 entries, 0 to 882
Data columns (total 10 columns):
 #   Column           Non-Null Count  Dtype  
---  ------           --------------  -----  
 0   title            426 non-null    object 
 1   location         426 non-null    object 
 2   country          426 non-null    object 
 3   employment_type  426 non-null    object 
 4   company_name     425 non-null    object 
 5   employee_qty     426 non-null    object 
 6   company_field    426 non-null    object 
 7   skills           426 non-null    object 
 8   job_description  426 non-null    object 
 9   applicants       319 non-null    float64
dtypes: float64(1), object(9)
memory usage: 36.6+ KB


### Проверка процента потерянных данных

Проверим какой процент данных был потерян в результате фильтрации. 

In [27]:
filtered_dataset.shape[0] / dataset.shape[0]

0.48081264108352145

Потери данных составили 52%. В рамках ТЗ - это допустимый показатель и свидетельствует о том, что в датасете содержалось достаточно много вакансий не связанных с анализом данных (разработка, менеджмент и прочее).

In [28]:
filtered_dataset.head()

title      location         country  \
0                     Data Analyst         Basel     Switzerland   
1         Data Analyst - Logistics      Coventry  United Kingdom   
2         Data Analyst - Logistics      Coventry  United Kingdom   
3  Data Analyst (Space & Planning)  South Molton  United Kingdom   
4                     Data Analyst        Lugano     Switzerland   

  employment_type                              company_name     employee_qty  \
0         On-site  PharmiWeb.Jobs: Global Life Science Jobs  11-50 employees   
1         On-site                      Resolute Recruitment    not specified   
2         On-site                      Resolute Recruitment    not specified   
3         On-site                       Mole Valley Farmers    not specified   
4         On-site                                   FORFIRM    not specified   

             company_field skills  \
0  Staffing and Recruiting   None   
1            not specified   None   
2            not specified   None   
3            not specified   None   
4            not specified   None   

                                     job_description  applicants  
0                              What You Will Achi...        47.0  
1                                                            NaN  
2                               Data Analyst - Lo...         NaN  
3                                Salary:     To b...         NaN  
4                                FORFIRM     is p...         NaN

### Изучение перечня скиллов

Посмотрим на перечень скиллов в описании к каждой вакансии, для этого воспользуемся методом unique()

In [29]:
filtered_dataset['skills'].unique()

array(['None',
       '<span class="visually-hidden"><!-- -->Skills: Анализ данных, Визуализация данных, +8 more<!-- --></span>',
       '<span class="visually-hidden"><!-- -->Skills: Панель управления, Анализ данных, +8 more<!-- --></span>',
       '<span class="visually-hidden"><!-- -->Skills: Data Analytics, Компьютерная наука, +8 more<!-- --></span>',
       '<span class="visually-hidden"><!-- -->Skills: SSRS, Услуги анализа, +8 more<!-- --></span>',
       '<span class="visually-hidden"><!-- -->Skills: Анализ данных, Аналитические навыки, +8 more<!-- --></span>',
       '<span class="visually-hidden"><!-- -->Skills: Статистическое моделирование, Моделирование данных, +8 more<!-- --></span>',
       '<span class="visually-hidden"><!-- -->Skills: Python, SQL, +8 more<!-- --></span>',
       '<span class="visually-hidden"><!-- -->Skills: Визуализация данных, Tableau, +8 more<!-- --></span>',
       '<span class="visually-hidden"><!-- -->Skills: Google Analytics, Анализ данных, +8 mor

Как видно из перечня скиллов, требования к вакансиям достаточно обоснованы. Часто встречается популярный для аналитиков стек: python, SQL, различные BI инструменты, а также более абстрактные требования: наука о данных, решение задач, математика и прочее.

В соответствии с ТЗ необходимо извлечь два основных скилла для вакансий. Воспользуемся методом str.split для разделения текстового массива и создадим столбец hs_1 в который запишем первый скилл из массива.

In [30]:
filtered_dataset['hs_1'] = filtered_dataset['skills'].str.split(":").str[1]

In [31]:
filtered_dataset['hs_1'].value_counts()

 Аналитические навыки, Ключевые показатели деятельности, +8 more<!-- --></span>                            27
 Data Analytics, Визуализация данных, +8 more<!-- --></span>                                               21
 Английский язык, Социальные медиа, +8 more<!-- --></span>                                                 14
 Визуализация данных, Data Analytics, +8 more<!-- --></span>                                                7
 Основные показатели, Математика, +8 more<!-- --></span>                                                    5
 Основные показатели, Scala, +8 more<!-- --></span>                                                         5
 Статистическое моделирование, Моделирование данных, +8 more<!-- --></span>                                 4
 Python, Аналитические навыки, +8 more<!-- --></span>                                                       4
 Анализ данных, Визуализация данных, +8 more<!-- --></span>                                                 3
 Моделиров

Как видно из полученных значений данные были спарсены с сайта linked in - поэтому подгружены значения из ячейки div (язык разметки страниц html). Воспользуемся методом split и еще раз разделим массив на несколько частей.

In [32]:
filtered_dataset['hs_top_1'] = filtered_dataset['hs_1'].str.split(",").str[0]

In [33]:
filtered_dataset.head()

title      location         country  \
0                     Data Analyst         Basel     Switzerland   
1         Data Analyst - Logistics      Coventry  United Kingdom   
2         Data Analyst - Logistics      Coventry  United Kingdom   
3  Data Analyst (Space & Planning)  South Molton  United Kingdom   
4                     Data Analyst        Lugano     Switzerland   

  employment_type                              company_name     employee_qty  \
0         On-site  PharmiWeb.Jobs: Global Life Science Jobs  11-50 employees   
1         On-site                      Resolute Recruitment    not specified   
2         On-site                      Resolute Recruitment    not specified   
3         On-site                       Mole Valley Farmers    not specified   
4         On-site                                   FORFIRM    not specified   

             company_field skills  \
0  Staffing and Recruiting   None   
1            not specified   None   
2            not specified   None   
3            not specified   None   
4            not specified   None   

                                     job_description  applicants hs_1 hs_top_1  
0                              What You Will Achi...        47.0  NaN      NaN  
1                                                            NaN  NaN      NaN  
2                               Data Analyst - Lo...         NaN  NaN      NaN  
3                                Salary:     To b...         NaN  NaN      NaN  
4                                FORFIRM     is p...         NaN  NaN      NaN

In [34]:
filtered_dataset['hs_top_1'].value_counts()

 Data Analytics                               37
 Аналитические навыки                         33
 Английский язык                              32
 Анализ данных                                23
 Python                                       19
 Наука о данных                               16
 Визуализация данных                          13
 Основные показатели                          11
 SQL                                          10
 Панель управления                             6
 Статистическое моделирование                  6
 VBA                                           6
 Microsoft Power BI                            6
 Моделирование данных                          4
 Управление данными                            4
 Решение задач                                 4
 Коммуникация                                  3
 Управление проектами                          3
 Microsoft Excel                               3
 Scala                                         3
 Математика         

Первый столбец с хард скиллами успешно сформирован.

In [35]:
filtered_dataset['hs_top_2'] = filtered_dataset['hs_1'].str.split(",").str[1]

In [36]:
filtered_dataset['hs_top_2'].value_counts()

 Ключевые показатели деятельности                     28
 Визуализация данных                                  25
 Data Analytics                                       21
 Аналитические навыки                                 16
 Анализ данных                                        16
 Социальные медиа                                     14
 Python                                               13
 SQL                                                  12
 Коммуникация                                          8
 Математика                                            8
 Решение задач                                         7
 Visualization                                         7
 Scala                                                 7
 Компьютерная наука                                    6
 Панель управления                                     6
 Наука о данных                                        6
 Моделирование данных                                  5
 Microsoft Excel               

Второй столбец с хард скиллами успешно сформирован.

Теперь удалим столбец hs_1 из которого получали данные по хард скиллам с помощью метода drop.

In [37]:
filtered_dataset.drop('hs_1', axis=1, inplace=True)

In [38]:
filtered_dataset.head()

title      location         country  \
0                     Data Analyst         Basel     Switzerland   
1         Data Analyst - Logistics      Coventry  United Kingdom   
2         Data Analyst - Logistics      Coventry  United Kingdom   
3  Data Analyst (Space & Planning)  South Molton  United Kingdom   
4                     Data Analyst        Lugano     Switzerland   

  employment_type                              company_name     employee_qty  \
0         On-site  PharmiWeb.Jobs: Global Life Science Jobs  11-50 employees   
1         On-site                      Resolute Recruitment    not specified   
2         On-site                      Resolute Recruitment    not specified   
3         On-site                       Mole Valley Farmers    not specified   
4         On-site                                   FORFIRM    not specified   

             company_field skills  \
0  Staffing and Recruiting   None   
1            not specified   None   
2            not specified   None   
3            not specified   None   
4            not specified   None   

                                     job_description  applicants hs_top_1  \
0                              What You Will Achi...        47.0      NaN   
1                                                            NaN      NaN   
2                               Data Analyst - Lo...         NaN      NaN   
3                                Salary:     To b...         NaN      NaN   
4                                FORFIRM     is p...         NaN      NaN   

  hs_top_2  
0      NaN  
1      NaN  
2      NaN  
3      NaN  
4      NaN

Задача по выделению хард скиллов была успешно выполнена.

### Изучение формата работы

Теперь посмотрим какое количество различных форматов работы присутствует в датасете. Воспользуемся методом value_counts()

In [39]:
filtered_dataset['employment_type'].value_counts()

Hybrid           198
On-site          143
Remote            47
not specified     38
Name: employment_type, dtype: int64

Самый популярный тип - гибрид (198 вакансий);  
Офис - 143 вакансии;  
Удаленная работа - 47 вакансий;  
Не указано - 38 вакансий  

### Формирование датасета для выгрузки

Выгрузим обработанный датасет в формат csv для дальнейшей работы с помощью BI инструментов.

In [40]:
filtered_dataset.to_csv('filtered_dataset.csv', index=False)

## Выводы по предобработке данных

**В результате преодобработки были проведены следующие действия:**  
1) Выполнена проверка на пропуски;  
2) Выполнена проверка на дубликаты, строки содержашие явные дубликаты были удалены;  
3) Датасет был отфильтрован таким образом, чтобы остались только вакансии Data Analyst и Data Scientist;  
4) Изучены основные особенности представленных данных;  
5) Изучен процент потерь данных после предобработки;  
6) Сформирован датасет для дальнейшей работы в BI инструментах.  

## Дашборд TableaU

Готовый дашборд доступен по ссылке: [Нажми для перехода](https://public.tableau.com/app/profile/sergei.v8049/viz/linked_in_vacancies_dashboard_ver_2_0/Dashboard1?publish=yes)